src/compiler/core.ts

```diff
@@ -1985,26 +1970,30 @@ namespace ts {
         let previous = new Array(s2.length + 1);
         let current = new Array(s2.length + 1);
         /** Represents any value > max. We don't care about the particular value. */
-        const big = max + 1;
+        const big = max + 0.000000001;
 
         for (let i = 0; i <= s2.length; i++) {
             previous[i] = i;
         }
 
         for (let i = 1; i <= s1.length; i++) {
             const c1 = s1.charCodeAt(i - 1);
-            const minJ = i > max ? i - max : 1;

```

Comment:
I think I understand the need for Math.ceil and Math.floor, but why change to Math.max and Math.min?

Reply:
`Math.max/min` is more readable for me. Anyway, reverted now.

Reply:
OK, makes sense. After this PR is in, maybe you can re-submit that change separately where it would be obvious that Math.max/min are equivalent. I thought it was more readable too, but I prefer to reduce code change per PR in complicated functions like this.